mcda/mcda_with_robustness.py

```diff
@@ -87,8 +89,14 @@ def create_n_randomly_sampled_matrices(self) -> List[pd.DataFrame]:
         input_matrix = self._input_matrix  # (AxnI)
         is_exact_pdf_mask = self.is_exact_pdf_mask
         is_poisson_pdf_mask = self.is_poisson_pdf_mask
+        random_seed = self.random_seed
 
-        np.random.seed(42)
+        default_random_seed = 42
+
+        if random_seed is not None:
+            np.random.seed(random_seed)
+        else:
+            np.random.seed(default_random_seed)
```

Comment:
Maybe just add `default_random_seed` as `42` here in `else` instead of declaring a variable before `if` as it is just used within  the `else` statement. 

or another option:
maybe just assign the value as `42` as before and just add `if` condition to re-assign value if `random_seed` is not None.